assignments/array-methods.js

```diff
@@ -56,25 +56,58 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
-console.log(fullName);
+
+runners.forEach((runners) => {
+
+    fullName.push(runners.first_name + runners.last_name);
+});
+
+console.log(fullName); 
+
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
+
+ var upperFirstName = runners.map((runner) => {
+ var myvariable = runner.first_name.toUpperCase();
+ allCaps.push(myvariable);  
+}); 
+
+console.log(allCaps);
+
 
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
+// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. 
+//Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
+
 let largeShirts = [];
+var sizeLarge = runners.filter((runners)=> {
```

Comment:
filter also creates a filter in place, so no need for if else statements within the filter or to push to a new array. Make sure you delete the empty largeShirts array on 84 too.
```suggestion
const largeShirts = runners.filter(runner => {
  return runner.shirt_size === "L"
})
```